### DISCUSSION TITLE
Exporting T1 Values

### DISCUSSION THREAD

**Discussion by AlexMorello** [qMRLab/qMRLab#461] [Q&A] [answered]
Dear devs, 

May I ask how the values obtained by placing the ROI markers in qMRLab are exported? I need to fit the T1 and T2 data in MATLAB. Trying to take [this approach,](https://yangweiliu.wordpress.com/2015/12/18/automatic-data-analysis-in-matlab-for-mri-and-nmr-t1t2-measurements/) but am kind of confused as to where I should begin. 

Thanks a lot in advance!

**Comment by AlexMorello** [comment]
Up

**Comment by mathieuboudreau** [comment, marked as answer]
Hi @AlexMorello,

Could you share a screenshot and explain the exact steps you've taken, so that we may try and reproduce what you're doing before proceeding?

**Reply by AlexMorello** [reply]
Thank you very much for responding so promptly, @mathieuboudreau!

Here's what I've got:
![Screenshot_2022-01-27_16-17-57](https://user-images.githubusercontent.com/84672824/151388233-33612b20-80e5-4f9b-8ca0-666a06e54b11.png)

I need to fit these mean values in MATLAB, but should I just write them down by hand for all the slices, or is there a way to export them somehow?

**Reply by agahkarakuzu** [reply]
On the upper right of the panel, there are colors ranged from 1 to 5 (red to purple).

Select one of them, right click one of the ROIs and select create mask. You will see that the color you selected will become the color of the mask. 

With ROI tool, you can have 5 region masks. Then, when you click export stats, it will give you summary per region. You can save them as well.

**Reply by mathieuboudreau** [reply]
>right click one of the ROIs and select create mask. 

It's actually "poly2mask", not "select create mask."

**Reply by mathieuboudreau** [reply]
To save, if you want to save the mask of ROIs as a NIFTI file, click on the blue "save" disk icon on the right panel (above the folder icon).
<img width="85" alt="Screen Shot 2022-01-27 at 12 40 23 PM" src="https://user-images.githubusercontent.com/1421029/151403416-7546594d-eb83-4623-bcd4-916e842fc8c6.png">

**Reply by AlexMorello** [reply]
> On the upper right of the panel, there are colors ranged from 1 to 5 (red to purple).
> 
> Select one of them, right click one of the ROIs and select create mask. You will see that the color you selected will become the color of the mask.
> 
> With ROI tool, you can have 5 region masks. Then, when you click export stats, it will give you summary per region. You can save them as well.

I see, thanks @agahkarakuzu @mathieuboudreau ! Where are the exported stats kept, though?

**Reply by agahkarakuzu** [reply]
> Where are the exported stats kept, though?

In the directory where you saved them.

**Reply by AlexMorello** [reply]
@agahkarakuzu I'm not sure I follow. I wasn't presented with a dialogue box to select the output directory upon clicking on 'export stats'.

**Reply by agahkarakuzu** [reply]
Can you please check your current (work) MATLAB directory? You will find the mat file with the name you exported.

**Reply by agahkarakuzu** [reply]
You can also see stats if you click the 'statistics' button under the 'Sanity check' text on the GUI. Color of the table(s) will be the same with that of roi(s).

**Reply by AlexMorello** [reply]
...it shows up in Workspace as a 1x1 struct instead

**Reply by mathieuboudreau** [reply]
<img width="859" alt="Screen Shot 2022-01-28 at 10 23 06 AM" src="https://user-images.githubusercontent.com/1421029/151563579-d2987099-a006-42dd-9eca-860c786dab64.png">
> Click "Statistics"
<img width="166" alt="Screen Shot 2022-01-28 at 10 23 15 AM" src="https://user-images.githubusercontent.com/1421029/151563590-7322e587-8595-4e12-be65-3b77ba9499db.png">
<img width="899" alt="Screen Shot 2022-01-28 at 10 23 31 AM" src="https://user-images.githubusercontent.com/1421029/151563620-9f408147-1019-4108-a721-d49b099a9ee3.png">
<img width="163" alt="Screen Shot 2022-01-28 at 10 23 49 AM" src="https://user-images.githubusercontent.com/1421029/151563638-7f7159ae-d011-4a98-92cf-c9eddb62e210.png">
<img width="853" alt="Screen Shot 2022-01-28 at 10 24 04 AM" src="https://user-images.githubusercontent.com/1421029/151563652-dc1b8ceb-b8e5-4847-b787-7f9fb7767b3f.png">
<img width="557" alt="Screen Shot 2022-01-28 at 10 24 38 AM" src="https://user-images.githubusercontent.com/1421029/151563661-2a02861e-34e3-4015-bfde-d1684a7ccfed.png">

**Reply by agahkarakuzu** [reply]
I thought you were referring to saving those vars, which you can by 

save myworkspace.mat

**Reply by AlexMorello** [reply]
Bravo and thanks @mathieuboudreau !

**Reply by AlexMorello** [reply]
> I thought you were referring to saving those vars, which you can by
> 
> save myworkspace.mat

No worries @agahkarakuzu  - now I know how to do both! Btw, is there some kind of user manual for this, just so that I wouldn't have to bother you folks all the time?

**Reply by mathieuboudreau** [reply]
>Btw, is there some kind of user manual for this, just so that I wouldn't have to bother you folks all the time?

This is our image viewer/ROI tool, more instructions can be found here: https://github.com/tanguyduval/imtool3D_td

If you have questions specific to the the image viewer/ROI tool, it may be best to raise issues there. The author was a previous qMRLab developer.

**Reply by AlexMorello** [reply]
I actually meant the entirety of qMRLab. Also, is it possible to export the values for the entire stack? Last but not least, can you keep the size of the ROI markers constant across the image? Will continue in that thread if you don't know.

**Reply by mathieuboudreau** [reply]
https://qmrlab.readthedocs.io/en/master/?version=latest

**Reply by mathieuboudreau** [reply]
>Also, is it possible to export the values for the entire stack? Last but not least, can you keep the size of the ROI markers constant across the image? 

I don't know the answers to these questions, they might be best to ask the imtool3D_td developpers.

**Reply by AlexMorello** [reply]
Thank you very much indeed once again!
